coderd/httpmw/apikey.go

```diff
@@ -346,15 +352,21 @@ func ExtractAPIKey(db database.Store, oauth *OAuth2Configs, redirectToLogin bool
 // apiTokenFromRequest returns the api token from the request.
 // Find the session token from:
 // 1: The cookie
-// 2: The old cookie
-// 3. The coder_session_token query parameter
-// 4. The custom auth header
+// 1: The devurl cookie
+// 3: The old cookie
+// 4. The coder_session_token query parameter
+// 5. The custom auth header
 func apiTokenFromRequest(r *http.Request) string {
 	cookie, err := r.Cookie(codersdk.SessionTokenKey)
 	if err == nil && cookie.Value != "" {
 		return cookie.Value
 	}
 
+	cookie, err = r.Cookie(DevURLSessionTokenCookie)
+	if err == nil && cookie.Value != "" {
+		return cookie.Value
+	}
+
```

Comment:
I'd move this after the header checks. If they pass a session token via the header, and a devurl cookie, the header value should be taken.